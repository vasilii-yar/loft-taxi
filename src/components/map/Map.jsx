import React from 'react';
import NavBar from "../navbar/NavBar";
import MapBox from "./MapBox";
import {connect} from "react-redux";
import RegistrationRedirectForm from "./RegistrationRedirectForm";
import RouteForm from "./RouteForm";
import {fetchProfileData} from "../../redux/modules/profile/profileActions";

class Map extends React.Component {

    componentDidMount() {
        this.props.fetchProfileData();
    }

    render() {
        const Form = this.props.isCardDataFilled ? RouteForm : RegistrationRedirectForm
        return (
            <>
                <header>
                    <NavBar/>
                </header>
                <main>
                    <section>
                        <MapBox/>
                        <Form/>
                    </section>
                </main>
            </>
        );
    }
}

const mapStateToProps = (state) => {
    return (
        {
            isCardDataFilled: state.profile.isCardDataFilled
        }
    )
}

export default connect(
    mapStateToProps,
    {fetchProfileData}
    )(Map);