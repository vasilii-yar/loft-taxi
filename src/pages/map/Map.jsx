import React from 'react';
import NavBar from "../../components/navbar/NavBar";
import MapBox from "../../components/mapbox/MapBox";

export default class Map extends React.Component {
    render() {
        return (
            <>
                <header>
                    <NavBar/>
                </header>
                <main>
                    <section>
                        <MapBox/>
                    </section>
                </main>
            </>
        );
    }
}