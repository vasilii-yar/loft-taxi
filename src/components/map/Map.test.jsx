import React from 'react';
import {render} from "@testing-library/react";
import Map from "./Map";
import renderer from "react-test-renderer";

jest.mock("./MapBox", () => (
    {
        __esModule: true,
        default: () => <div>Карта</div>
    }
));
jest.mock("../navbar/NavBar", () => (
    {
        __esModule: true,
        default: () => <div>NavBar component</div>
    }
));
jest.mock('mapbox-gl/dist/mapbox-gl', () => ({
    Map: () => ({})
}));

describe("Map", () => {
    it("renders correctly", () => {
        const {container} = render(
            <Map />
        );
        expect(container.innerHTML).toMatch("Карта");
    })

    it("renders correctly check by snapshot", () => {
        const tree = renderer.create(
            <Map />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    })
});