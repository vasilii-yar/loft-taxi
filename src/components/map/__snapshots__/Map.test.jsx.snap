// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map renders correctly check by snapshot 1`] = `
Array [
  <header>
    <div>
      NavBar component
    </div>
  </header>,
  <main>
    <section>
      <div>
        Карта
      </div>
      <div
        className="MuiPaper-root redirect-form-layout MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          className="MuiBox-root MuiBox-root-1"
        >
          <form
            className="redirect-form-content"
            onSubmit={[Function]}
          >
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column"
            >
              <div
                className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
              >
                <h4
                  className="MuiTypography-root MuiTypography-h4"
                >
                  Заполните платежные данные
                </h4>
              </div>
              <div
                className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
              >
                <p
                  className="MuiTypography-root MuiTypography-body1"
                >
                  Укажите информацию о банковской карте, чтобы сделать заказ.
                </p>
              </div>
              <div
                className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
              >
                <div
                  className="redirect-button"
                >
                  <button
                    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
                    disabled={false}
                    onBlur={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={0}
                    type="submit"
                  >
                    <span
                      className="MuiButton-label"
                    >
                      Перейти в профиль
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </form>
        </div>
      </div>
    </section>
  </main>,
]
`;
